code/obj/item/grenades.dm

```diff
@@ -126,19 +126,24 @@ ABSTRACT_TYPE(/obj/item/old_grenade/spawner)
 	sound_armed = 'sound/weapons/armbomb.ogg'
 	is_dangerous = FALSE
 	var/payload = null
+	var/amount_to_spawn = 5
 
 	prime()
 		var/turf/T = ..()
 		if (T)
 			playsound(T, 'sound/weapons/flashbang.ogg', 25, 1)
 			new payload(T)
-			for (var/i = 1; i<= 8; i= i*2)
-				if (istype(get_turf(get_step(T,i)),/turf/simulated/floor))
-					new payload(get_step(T,i))
+			var/dir_id = 1
+			for (var/i in 2 to src.amount_to_spawn)
+				var/turf/adjacent = get_step(T, cardinal[dir_id])
+				if (istype(adjacent,/turf/simulated/floor))
+					new payload(adjacent)
 				else
 					new payload(T)
+				dir_id++
+				if (dir_id > 4)
+					dir_id = 1
```

Comment:
Would prefer if this was a `%` and operating on length of the list instead of a magic number.

Reply:
Er yeah, not sure why I wrote it like that actually, fixed!